reframe/core/modules.py

```diff
@@ -1002,41 +997,38 @@ def available_modules(self, substr):
         return ret
 
     def _module_full_name(self, name):
-        return self.execute('find', '--format', '{name}-{version}-{hash}',
+        return self.execute('find', '--format', self._name_format,
                             name)
 
     def loaded_modules(self):
-        try:
-            hashes = '/'.join(
-                [m for m in os.environ['SPACK_LOADED_HASHES'].split(':') if m])
-            if hashes:
-                output = self.execute('find', '--loaded', '--format',
-                                      '{name}-{version}-{hash}', f'/{hashes}')
-                return [Module(m) for m in output.split('\n') if m]
-            else:
-                return []
-        except KeyError:
+        hashes = '/'.join(
+            [m for m in os.environ['SPACK_LOADED_HASHES'].split(':') if m])
+        if hashes:
+            output = self.execute('find', '--loaded', '--format',
+                                  self._name_format, f'/{hashes}')
```

Comment:
This produces this output `/hash1/hash2/.../hashn`. Is this valid? @victorusu ?

Reply:
Yes.  Module names can be passed to Spack as hashes, the syntax is `/<hash>`. For instance, this converts from hashes, which is how they are stored on the `SPACK_LOADED_HASHES` variable, to module names:
```
$> spack find --loaded --format '{name}/{version}' /vcsrq3vgchevv2weu3cfwe37jclspuuv/injxlcmccya6nslhqf7chwsh4e5hpgcp/zeyd2m5zugdxbnx5mepvbmfnht4tjrqr/oxpxbrpqqca5hjh6blb4qv5chqqv3ykv/sl7m27mzkbejtkrajigj3a3m37ygv4u2/4kcnj3oypwyyr3o46ipejwuk3x5gzrar/af5tdk6ilv6mah2ntgb5odryvlosijnz/fueeuyvb37exbhwxkgte4kurdjyxli5u
hwloc/1.11.11  libpciaccess/0.16  numactl/2.0.14  xz/5.2.5
libiconv/1.16  libxml2/2.9.10     openmpi/3.1.6   zlib/1.2.11
```
It doesn't need spaces between hashes.

Reply:
I just realized that the env variable is no needed, I think that was the initial way I was doing it. With `--loaded` it gives the loaded modules directly.